distributed/deploy/tests/test_local.py

```diff
@@ -411,27 +423,38 @@ async def test_memory_limit_none():
         assert w.memory_manager.memory_limit is None
 
 
-def test_cleanup():
-    with clean(threads=False):
-        c = LocalCluster(n_workers=2, silence_logs=False, dashboard_address=":0")
-        port = c.scheduler.port
-        c.close()
-        c2 = LocalCluster(
-            n_workers=2, scheduler_port=port, silence_logs=False, dashboard_address=":0"
-        )
-        c2.close()
+def test_cleanup(loop):
+    c = LocalCluster(n_workers=2, silence_logs=False, dashboard_address=":0", loop=loop)
+    port = c.scheduler.port
+    c.close()
+    c2 = LocalCluster(
+        n_workers=2,
+        scheduler_port=port,
+        silence_logs=False,
+        dashboard_address=":0",
+        loop=loop,
+    )
+    c2.close()
 
 
-def test_repeated():
-    with clean(threads=False):
-        with LocalCluster(
-            n_workers=0, scheduler_port=8448, silence_logs=False, dashboard_address=":0"
-        ):
-            pass
-        with LocalCluster(
-            n_workers=0, scheduler_port=8448, silence_logs=False, dashboard_address=":0"
-        ):
-            pass
+def test_repeated(loop_in_thread):
+    loop = loop_in_thread
```

Comment:
What makes you need loop_in_thread instead of just loop?

Reply:
the `LocalCluster.__exit__` closes the loop that's passed in - unless it's already running in another thread